### PR TITLE
Fix Challange and Riddle handling bad responses

### LINKED ISSUE #194: [BUG] - Riddle and Challenge command improperly handle bad responses
# Bug Report

## Describe the bug

The APIs for these commands are designed to return a JSON object with an `error` property when a user provides the wrong answer. The `challenge` command handles this in a `catch` statement while the `riddle` command crashes. Neither of these commands are properly displaying the response to the user.

## Expected behavior

Both commands need to be adjusted to provide the feedback to the user regarding their command. Additionally, the `riddle` command needs to take the full answer (instead of using `.shift()`).

### PR DESCRIPTION
**Pull request by Andreybest**
## Description:

Fixed exception on riddle command, and added stylized error messages to users on challange and riddle command.

Also changed forming answer from one args to full sentence in answer to a riddle.

## Related Issue:

Closes #194

### CODE REVIEW

File: src/commands/games/challenge.ts [hunk 1]
```diff
@@ -159,11 +159,14 @@ const challenge: CommandInt = {
         error?.response?.data?.message ?? "Unknown error."
       );
 
-      await message.reply(
-        `Sorry, but I cannot execute the \`${action}\` action inside the challenge.`
+      const errorEmbed = new MessageEmbed();
+      errorEmbed.setColor(bot.color);
+      errorEmbed.setTitle("Challenge Error");
+      errorEmbed.setDescription(
+        "Error happened while working with your challenge. Please, try again, or wait a bit."
```
Comment (nhcarrigan) [inline src/commands/games/challenge.ts:166]: So this does not actually fix the issue - it sends an embed on error, but the challenge API actually sends back a JSON error object if the user's answer is incorrect. The message, contained in `error.response.data.message`, needs to be sent back to the user. It's currently being logged to the bot console instead.

File: src/commands/games/riddle.ts [hunk 2]
```diff
@@ -35,100 +35,116 @@ const riddle: CommandInt = {
...
+
+      const errorEmbed = new MessageEmbed();
+      errorEmbed.setColor(bot.color);
+      errorEmbed.setTitle("Riddle Error");
+      errorEmbed.setDescription(
+        "Error happened while working on your riddle. Please, try again, or wait a bit."
+      );
+
+      await message.reply(errorEmbed);
+    }
```
Comment (nhcarrigan) [inline src/commands/games/riddle.ts:147]: Same as my other comment.

**Review by nhcarrigan — CHANGES_REQUESTED** [review]

File: src/commands/games/challenge.ts (hunk 1 above)
Comment (Andreybest) [reply src/commands/games/challenge.ts:166]: Sorry, I thought that text would be enough, but you've right, we need to send info from server. Im working on it rn.

**Comment by Andreybest** [comment]
I've added handling in case if user gives wrong answer in challenge or riddle. Wrong answers are responded with 400, and the bot reponds with "Result - incorrect" and returns message from response.

**Review by nhcarrigan — CHANGES_REQUESTED** [review]
This will ensure the challenge's response gets sent back to the user correctly.

**Comment by Andreybest** [comment]
Sure!

**Review by nhcarrigan — APPROVED** [review]
LGTM~! 🥳 

This should correct the issue.

**Pull request merged**